portal-test/src/com/liferay/portal/kernel/test/util/TestPropsUtil.java

```diff
@@ -49,7 +50,13 @@ private TestPropsUtil() {
 			ReflectionUtil.throwException(ioException);
 		}
 
-		try (InputStream inputStream = TestPropsUtil.class.getResourceAsStream(
+		ClassLoader classLoader = PortalClassLoaderUtil.getClassLoader();
+
+		if (classLoader == null) {
+			classLoader = TestPropsUtil.class.getClassLoader();
```

Comment:
Rather than fallback to current class loading classloader.The test rule should make sure portal classloader is always set.

For integration test, this should already be set by portal. Therefore I can only see unit test may run into this fallback.
The fix should be adding in a PortalClassLoaderUtil.setClassLoader() call inside InitializeKernelUtilTestRule, rather than making TestPropsUtil doing this.